### PR TITLE
Parameterize tests with different domain home source types

### PR DESCRIPTION
**Pull request by xiancao**
Parameterize the tests to run against any domain home source type such as ModelInImage, DomainInPV and DomainInImage.

removed ItDomainInImageWDT.java
removed testDomainInPvUsingWdt() in ItDomainInPV.java
moved testLivenessProbe() in ItDomainInPV.java to ItParameterizedDomain.java

Jenkins test result:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/984/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/985/

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 1]
```diff
@@ -751,6 +750,22 @@ public static void checkServiceExists(String serviceName, String namespace) {
...
+   * @param domainUid the label the pod is decorated with
+   * @param namespace the namespace in which the pod exists
+   */
+  public static void checkPodReadyAndServiceExists(String podName, String domainUid, String namespace) {
+    LoggingFacade logger = getLogger();
+    logger.info("Waiting for pod {0} to be ready in namespace {1}", podName, namespace);
+    checkPodReady(podName, domainUid, namespace);
+
+    logger.info("Check service {0} exists in namespace {1}", podName, namespace);
+    checkServiceExists(podName, namespace);
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:766]: The service check should be done before pod ready check.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 2]
```diff
@@ -0,0 +1,1284 @@
...
+
+      try {
+        Thread.sleep(2 * 1000);
+      } catch (InterruptedException ie) {
+        // ignore
+      }
+    }
+
+    // check pod is ready
+    checkPodReady(serverName, domainUid, domainNamespace);
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:425]: Does the service check should be done?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 3]
```diff
@@ -0,0 +1,1284 @@
...
+    assertNotNull(namespaces.get(3));
+    String domainInPVNamespace = namespaces.get(3);
+    assertNotNull(namespaces.get(4));
+    String domainInImageNamespace = namespaces.get(4);
+
+    // set the service account name for the operator
+    opServiceAccount = opNamespace + "-sa";
+
+    // get a free port for external REST HTTPS port
+    externalRestHttpsPort = getNextFreePort(31001, 31201);
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:205]: You can use 0 for external rest port as well. I did the same in upgrade tests - https://github.com/oracle/weblogic-kubernetes-operator/blob/operator-upgrade-tests/new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 4]
```diff
@@ -0,0 +1,1284 @@
...
+    assertTrue(callWebAppAndWaitTillReady(curlCmd, 60));
+    logger.info("WebLogic console on domain1 is accessible");
+  }
+
+  /**
+   * Verify liveness probe by killing managed server process 3 times to kick pod container auto-restart.
+   */
+  @ParameterizedTest
+  @DisplayName("Test liveness probe of pod")
+  @MethodSource("domainProvider")
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:360]: can you test this for only one type of domain, no need to test with all three models, just adds up run time.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 3 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:205]: you can make this change in your next PR if you want.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 5]
```diff
@@ -0,0 +1,1285 @@
...
+                .addEnvItem(new V1EnvVar()
+                    .name("JAVA_OPTIONS")
+                    .value("-Dweblogic.StdoutDebugEnabled=false "
+                        + "-Dweblogic.debug.DebugDiagnosticsExpressionFunctionMapper=true "
+                        + "-Dweblogic.StdoutDebugEnabled=true -Dweblogic.log.LogSeverity=Debug "
+                        + "-Dweblogic.log.LoggerSeverity=Debug -Dweblogic.debug.DebugDiagnosticsUtils=true "
+                        + "-Dweblogic.debug.DebugDiagnosticsExpressionFunctions=true "
+                        + "-Dweblogic.debug.DebugDiagnosticsExpressionPoller=true "
+                        + "-Dweblogic.debug.DebugDiagnosticWatch=true"))
+                .addEnvItem(new V1EnvVar()
```
Comment (marinakog) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:1199]: do we need so many debug flags here?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 6]
```diff
@@ -2211,7 +2238,7 @@ public static void createDomainJob(String image, boolean isUseSecret, String pvN
                     .containers(Arrays.asList(jobContainer  // container containing WLST or WDT details
                         .name("create-weblogic-domain-onpv-container")
                         .image(image)
-                        .imagePullPolicy("Always")
+                        .imagePullPolicy("IfNotPresent")
                         .ports(Arrays.asList(new V1ContainerPort()
```
Comment (marinakog) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:2242]: would it work on multinoded cluster? why switch from Always?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 2 above)
Comment (xiancao) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:426]: fixed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 3 above)
Comment (xiancao) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java]: fixed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 4 above)
Comment (xiancao) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java]: fixed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 5 above)
Comment (xiancao) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:1181]: fixed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java (hunk 6 above)
Comment (xiancao) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java]: fixed

**Pull request merged**